apply patch from kokkos/mdspan#425

ndellingwood opened the pull request:
fix typo in layout_mapping_alike concept (#425)

detected during testing with intel/2023.2 with c++20

ndellingwood committed:
apply patch from kokkos/mdspan#425

fix typo in layout_mapping_alike concept (#425)

detected during testing with intel/2023.2 with c++20

Signed-off-by: Nathan Ellingwood <ndellin@sandia.gov>

tpadioleau reviewed (changes requested):
Can you update the commit sha in `.github/workflows/mdspan-version-check.yml` to refer to your commit in the mdspan repo https://github.com/kokkos/mdspan/commit/c133f7ac5e4d2b9db2fe80f5944fff98b5780758 ?

ndellingwood committed:
update mdspan sha

update to c133f7ac5e4d2b9db2fe80f5944fff98b5780758 correspnding to
kokkos/mdspan#425

Thanks to @tpadioleau

Signed-off-by: Nathan Ellingwood <ndellin@sandia.gov>

ndellingwood commented:
> Can you update the commit sha in `.github/workflows/mdspan-version-check.yml` to refer to your commit in the mdspan repo [kokkos/mdspan@c133f7a](https://github.com/kokkos/mdspan/commit/c133f7ac5e4d2b9db2fe80f5944fff98b5780758) ?

@tpadioleau updated in 9bea3a004cc5221c126a8d16237e0a4b6b78ebcc , thanks for the suggestion!

ndellingwood commented:
Still failing the mdspan check, looks like a full mdspan snapshot is required, rather than cherry-picking over a specific fix?

tpadioleau commented:
> Still failing the mdspan check, looks like a full mdspan snapshot is required, rather than cherry-picking over a specific fix?

That is how the test works indeed. Is it an issue ?

ndellingwood commented:
@tpadioleau yes, I only cherry-picked a small typo fix patch here and assumed that was valid. I think there is ongoing broader discussion on updating the full snapshot of mdspan [here](https://github.com/kokkos/kokkos/pull/8320) , maybe that PR can be updated to incorporate these changes

masterleinad commented:
Superseded by #8320.
